cmd/server/iwf/iwf.go

```diff
@@ -110,21 +111,35 @@ func start(c *cli.Context) {
 	// The client is a heavyweight object that should be created once per process.
 	var unifiedClient uclient.UnifiedClient
 	if config.Interpreter.Temporal != nil {
+		temporalConfig := config.Interpreter.Temporal
+
 		var metricHandler client.MetricsHandler
-		if config.Interpreter.Temporal.Prometheus != nil {
-			pscope := newPrometheusScope(*config.Interpreter.Temporal.Prometheus, logger)
+		if temporalConfig.Prometheus != nil {
+			pscope := newPrometheusScope(*temporalConfig.Prometheus, logger)
 			metricHandler = sdktally.NewMetricsHandler(pscope)
 		}
 
-		temporalClient, err := client.Dial(client.Options{
-			HostPort:       config.Interpreter.Temporal.HostPort,
-			Namespace:      config.Interpreter.Temporal.Namespace,
-			MetricsHandler: metricHandler,
-		})
+		var temporalClient client.Client
+		var err error
+		if temporalConfig.CloudAPIKey != "" {
+			input := TemporalCloudNamespaceClientInput{
+				Namespace:      temporalConfig.Namespace,
+				Auth:           &ApiKeyAuth{APIKey: temporalConfig.CloudAPIKey},
+				MetricsHandler: metricHandler,
+			}
+			temporalClient, err = GetTemporalCloudNamespaceClient(context.Background(), &input)
+		} else {
+			temporalClient, err = client.Dial(client.Options{
+				HostPort:       temporalConfig.HostPort,
+				Namespace:      temporalConfig.Namespace,
+				MetricsHandler: metricHandler,
+			})
+		}
+
 		if err != nil {
 			rawLog.Fatalf("Unable to connect to Temporal because of error %v", err)
```

Comment:
Can we change this error message to be conditional of the connection type we're doing? i.e. 

```
if err != nil {
    if temporalConfig.CloudAPIKey != "" {
        rawLog.Fatalf("Unable to connect to Temporal Cloud (namespace: %s) because of error: %v", temporalConfig.Namespace, err)
    } else {
        rawLog.Fatalf("Unable to connect to Temporal because of error %v", err)
    }
}
```

Reply:
that's a good idea. However, I have reverted the change to use cloud SDK. The current temporal goSDK does support API key out of the box.  And the cloud SDK is still experiemental.